runtime/lua/vim/treesitter.lua

```diff
@@ -96,4 +98,38 @@ function M.get_string_parser(str, lang, opts)
   return LanguageTree.new(str, lang, opts)
 end
 
+local function setup_highlight_for_buf(bufnr)
+    -- Setup treesitter highlighting for new buffer
+    local ft = a.nvim_buf_get_option(bufnr, "ft")
+    if not pcall(language.require_language, ft) then return end
+
+    local tree = M.get_parser(bufnr, ft)
+    M.highlighter.new(tree)
```

Comment:
as an option, I would probably expect `treesitter` to not influence `syntax`. But it currently does, maybe this could be change within this PR. So you can have `treesitter=hl`/`syntax=on` or `treesitter=hl`/`syntax=off`, but with syntax only explicitly changing. `syntax=off,on,treesitter` is probably also confusing.

Reply:
I think allowing both is the right way since `:syntax` does provide a simple way of extending syntax highlighting. For example I use it for a bunch of niche filetypes where I want to highlight something specific to a project. I could migrate these to use TS queries, but why bother.

Reply:
still, `:set treesitter=hl` should not cause `syntax=off`